### ISSUE TITLE
Prohibit focus on disabled input element

### ISSUE THREAD

**Issue by palpatine1991** [DevExpress/testcafe#2037]
### Are you requesting a feature or reporting a bug?
Bug report

### What is the current behavior?
In real environment when you set focus on disabled input by javascript, the "focusin" is not dispatched. On the other hand, in testcafe environment, the event is dispatched.

### What is the expected behavior?
It should behave the same as in reality (without patchworked native events).

### How would you reproduce the current behavior (if this is a bug)?
Sorry for using angular code, but it is much easier to write the example (if you do not know angular, let me know, I can explain it more):

Template: 
```
<button id="focusButton" (click)="getFocusNative()"></button>
<input #inputNative (focusin)="focusIn()" disabled value="Google">
```

Component class:
```
@ViewChild('inputNative')
  inputNative:ElementRef;

getFocusNative() {
  this.inputNative.nativeElement.focus();
}

focusIn() {
  alert("Im focused!");
}
```

testcafe code:
```
test('Should do nothing when click on disabled material input', async () => {
    await t
        .click(Selector('#focusButton'));
});
```

In real enviroment, nothing happens. On the other hand in testcafe it opens an alert box.
### Specify your

* operating system: Windows 7
* testcafe version: 0.18.5
* node.js version: 6.10.0

**Comment by AndreyBelym** [comment]
Hello @palpatine1991! Thank you for the feedback. I've reproduced the issue in Chrome, but It should be checked in other popular browsers as well. There is a lot of differences in focus systems of different browsers.

**Comment by LavrovArtem** [comment]
The fix is published in `testcafe@0.19.1-dev20180305`
